commands/cfg/whoami.go

```diff
@@ -0,0 +1,90 @@
+package cfg
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/ionos-cloud/ionosctl/v6/internal/jwt"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/config"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
+
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/spf13/viper"
+
+	"github.com/ionos-cloud/ionosctl/v6/pkg/core"
+)
+
+func WhoamiCmd() *core.Command {
+
+	cmd := core.NewCommand(context.Background(), nil, core.CommandBuilder{
+		Verb:      "whoami",
+		ShortDesc: "Tells you who you are logged in as. Use `--provenance` to debug where your credentials are being used from",
+		LongDesc: fmt.Sprintf(`This command will tell you the email of the user you are logged in as.
+You can use '--provenance' flag to see which of these sources are being used. Note that If authentication fails, this flag is set by default.
+If using a token, it will use the JWT's claims payload to find out your user UUID, then use the Users API on that UUID to find out your e-mail address.
+If no token is present, the command will fall back to using the username and password for authentication.
+
+%s`, constants.DescAuthenticationOrder),
+		Example: `ionosctl cfg whoami
+ionosctl cfg whoami --provenance`,
+		PreCmdRun: core.NoPreRun,
+		CmdRun: func(c *core.CommandConfig) error {
+			cl, err := client.Get()
+
+			if fn := core.GetFlagName(c.NS, constants.FlagProvenance); err != nil || viper.GetBool(fn) {
+				return handleProvenance(c, cl, err)
+			}
+
+			// - - - Below this point, we are sure that client.Get() has returned a valid client object.
+
+			if !cl.IsTokenAuth() {
+				// Handle Username & Password Authentication
+				_, err = fmt.Fprintln(c.Command.Command.OutOrStdout(), cl.CloudClient.GetConfig().Username)
+				return err
+			}
+
+			// Handle token authentication
+			usernameViaToken, jwtParseErr := jwt.Username(cl.CloudClient.GetConfig().Token)
+			if jwtParseErr != nil {
+				return fmt.Errorf("failed getting username via token: %w", err)
```

Comment:
err, or jwtParseErr?

Reply:
yes `jwtParseErr`.

`err` is handled above - it forces `--provenance` if getting credentials fails.

However, I think I will rename `err` to `authErr` for clarity.

Reply:
ah, sorry, now I see that you meant the question for the embedded error message in `Errorf` call - and not the `if` statement.

Yes, here should be `jwtParseErr` too - good catch